tests/test_image.py

```diff
@@ -0,0 +1,260 @@
+                {"type": "PupilAnnulusSampler", "R_outer": 4.18, "R_inner": 2.55},
+                {
+                    **optics_args,
+                    "boresight": boresight,
+                    "camera": "LsstCam",
+                },
+            ]
+        },
+    }
+    galsim.config.ProcessInput(config)
+    return config
+
+
+def run_lsst_image(image_type):
+    """Create an image of the given type and check that objects are batched as expected and stars at the correct positions."""
+
+    config = create_test_config(image_type)
+
+    n_images = 1
+    n_expected_objects = 20
+    all_obj_indices = frozenset(range(20))
+    expected_fft_obj_indices = frozenset(range(10))
+    with assert_no_error_logs(logger_level=logging.INFO) as logger:
+        [image] = galsim.config.BuildImages(n_images, config, logger=logger)
+    image.write("/tmp/tiny_instcat.fits")
+    fft_obj_indices = {n for n in range(n_expected_objects) if any(f"Use FFT for object {n}." in msg for msg in logger.messages)}
+    phot_obj_indices = {n for n in range(n_expected_objects) if any(f"Use photon shooting for object {n}." in msg for msg in logger.messages)}
+    assert fft_obj_indices == expected_fft_obj_indices
+    assert phot_obj_indices == all_obj_indices - expected_fft_obj_indices
+
+    # Objects in the same order as in the catalog. Positions of [0, 0] mean
+    # That the objects are outside of the image.
+    expected_positions = np.array([
+        [2046, 2000],
+        [3770, 1128],
+        [0, 0],
+        [3777, 3880],
+        [3817, 3934],
+        [3727, 1123],
+        [0, 0],
+        [3722, 1119],
+        [3721, 1094],
+        [3752, 1124],
+        [3702, 1081],
+        [3762, 1137], # Obj #11
+        [3715, 1114],
+        [3762, 1135], # Almost indistinguishable from #11
+        [3789, 1099],
+        [3802, 3273],
+        [3898, 3275],
+        [3969, 3292],
+        [3482, 3339],
+        [3452, 3751],
+    ])
+    expected_brightness_values = np.array([
+        1.974843e+06,
+        4.325211e+06,
+        0.0,
+        1.978054e+06,
+        1.973144e+06,
+        1.981440e+06,
+        0.0,
+        3.131857e+06,
+        1.980739e+06,
+        1.977764e+06,
+        1.983800e+05,
+        4.045435e+06,
+        2.195470e+06,
+        4.045435e+06,
+        1.974780e+05,
+        1.966890e+05,
+        1.968660e+05,
+        1.970680e+05,
+        1.965050e+05,
+        4.000000e+01,
+    ])
+    assert_objects_at_positions(image.array, expected_positions, expected_brightness_values)
+
+
+def test_lsst_image_original_pipeline():
+    """Check that LSSTImage batches objects as expected and renders objects at the correct positions."""
+    run_lsst_image("LSST_Image")
+
+def test_lsst_image_photon_pooling_pipeline():
+    """Check that LSST_PhotonPoolingImage batches objects as expected and renders objects at the correct positions."""
+    run_lsst_image("LSST_PhotonPoolingImage")
```

Comment:
I feel like we probably want a few additional tests in this pipeline.  E.g. that the FFT and faint objects only got drawn once.  And the others got drawn nbatch times.

Reply:
I've written a set of tests for the partitioning and batching process. As mentioned above, it also confirms that total flux is conserved for photon objects,